Documentation/ExtensionArchitecture/Extbase/Introduction/Index.rst

```diff
@@ -0,0 +1,93 @@
+.. include:: /Includes.rst.txt
+.. index::
+   pair: Extension development; Extbase introduction
+
+.. _extbase-introduction:
+
+====================
+Extbase introduction
+====================
+
+.. todo:: Refer to a good example, e.g. tea, blog_example?
+
+.. todo:: Add tips how to get more information or help
+
+What is Extbase?
+================
+
+Extbase is provided via a TYPO3 system extension (typo3-cms/extbase). Extbase is
+a framework within the TYPO3 Core. The framework is based on the software pattern
+`MVC (model-view-controller) <https://en.wikipedia.org/wiki/Model%E2%80%93view%E2%80%93controller>`__
+and uses
+`ORM (object relational modeling) <https://en.wikipedia.org/wiki/Object%E2%80%93relational_mapping>`__.
+
+Extbase can be and is often used in combination with the Fluid templating engine,
+but Fluid **can** also be used without Extbase. All things that can be done with
+Extbase, can also be done with TYPO3 Core native functionality.
+Extbase is not a prerequisite for extension development. In most cases, not
+using Extbase means writing more code. In many cases the performance will be
+better when not using Extbase.
+
+When Extbase was released, it was introduced as the modern way to program
+extensions and the "old" way (pibase) was propagated as outdated. When we look
+at this today, it is not entirely true: Extbase is a good fit for some specific
+types of extensions and there are always alternatives. For some use cases it
+is not a good fit at all and the extension can and should be developed without
+Extbase.
+
+Thus, many things, such as backend modules or plugins can be done "the Extbase
+way" or "the Core way". This is a design decision, the extension developer must
+make for the specific use case.
+
+.. important::
+
+   Extbase is one of many options in the "toolbox" of TYPO3 extension development.
+
+   TYPO3 is flexible, often provides several approaches for one thing and it is
+   up to the extension developer to make an informed choice.
+
+
+Extbase or not?
+===============
+
+When to use Extbase and when to use other methods?
+
+As a rule of thumb (which should not be blindly followed but gives some
+guidance):
+
+Use Extbase if you:
```

Comment:
If you want to benefit from the the "Community" approach of developing extensions (f.e. sharing knowledge, find contributors)

Reply:
@sbusemann 
> If you want to benefit from the the "Community" approach of developing extensions (f.e. sharing knowledge, find contributors)

I am not sure what you mean. Is the "sharing knowledge", "finding contributors" not independent of using Extbase or not?

Reply:
Extbase is a kind of standard for extensions within the TYPO3 community. It is easier to get help, contributors if an author follows a standard. Following a non standard way, might be more performant, but it could lead to the situation, what we had in the past, "spaghetti code" and 1000 different approaches.

Reply:
Mind you, I have seen serious spaghetti code with Extbase too. And some developers have a panache for seriously bad domain driven design. 😆 Best practices and standard approaches are important independently of the architecture.

Reply:
maybe we can leave that point for now, if there is more to be debated. Can be added later.